Formplayer/testPages/app_preview/app_preview_basics.py

```diff
@@ -0,0 +1,236 @@
+        self.wait_to_click(self.select_test_application)
+        self.wait_to_click(self.view_app_preview)
+        self.switch_to_frame(self.iframe)
+        self.wait_to_click(self.start_option)
+        time.sleep(5)
+        assert not self.is_present_and_displayed(self.start_option)
+        time.sleep(5)
+        self.switch_to_default_content()
+        self.wait_to_click(self.back_button)
+        self.switch_to_frame(self.iframe)
+        assert self.is_present_and_displayed(self.start_option)
+        print("Back button works!!")
+
+    def refresh_button_functionality_01(self):
+        self.wait_to_click(self.application_menu_id)
+        self.wait_to_click(self.select_test_application)
+        self.wait_to_click(self.view_app_preview)
+        self.switch_to_frame(self.iframe)
+        self.wait_to_click(self.start_option)
+        self.wait_to_click(self.case_list_menu)
+        self.switch_to_default_content()
+        time.sleep(5)
+        self.wait_to_click(self.refresh_button)
+        time.sleep(3)
+        self.switch_to_frame(self.iframe)
+        # time.sleep(3)
+        # assert not self.is_present_and_displayed(self.registration_form)
+        assert self.is_present_and_displayed(self.start_option)
+        print("Refresh Button works!!")
+
+    def refresh_button_functionality_02(self):
+        self.wait_to_click(self.application_menu_id)
+        self.wait_to_click(self.select_test_application)
+        self.wait_to_click(self.view_app_preview)
+        self.wait_to_click(self.form_builder_registration_form)
+        self.wait_to_click(self.add_question)
+        self.wait_to_click(self.text_question_type)
+        self.wait_to_clear_and_send_keys(self.text_question_display_text, "Test")
+        self.wait_to_click(self.save_question_button)
+        # time.sleep(5)
+        self.wait_to_click(self.refresh_button)
+        self.switch_to_frame(self.iframe)
+        self.wait_to_click(self.start_option)
+        self.wait_to_click(self.case_list_menu)
+        self.wait_to_click(self.registration_form)
+        assert self.is_present_and_displayed(self.new_test_question)
+        self.switch_to_default_content()
+        time.sleep(3)
+        self.wait_to_click(self.formplayer_test_question)
+        self.wait_to_click(self.delete_button)
+        self.wait_to_click(self.save_question_button)
+        time.sleep(3)
+        self.wait_to_click(self.refresh_button)
+        self.switch_to_frame(self.iframe)
+        self.wait_to_click(self.start_option)
+        self.wait_to_click(self.case_list_menu)
+        self.wait_to_click(self.registration_form)
+        assert not self.is_present_and_displayed(self.new_test_question)
+        print("Refresh button works!!")
+
+    def web_user_submission(self):
+        self.wait_to_click(self.application_menu_id)
+        self.wait_to_click(self.select_test_application)
+        self.wait_to_click(self.view_app_preview)
+        self.switch_to_frame(self.iframe)
```

Comment:
```
self.wait_to_click(self.application_menu_id)
self.wait_to_click(self.select_test_application)
self.wait_to_click(self.view_app_preview)
```
the above code is reused in multiple functions here. So create a function as 
` def open_view_app_preview(self):
        self.wait_to_click(self.application_menu_id)
        self.wait_to_click(self.select_test_application)
        self.wait_to_click(self.view_app_preview)`
so that we can directly call it in our testCases as
`app_preview = AppPreviewBasics(driver)
    app_preview.open_view_app_preview()`